# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../kdebindings/ruby_const-fix.patch
# Copyright (C) 2008 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Index: kdebindings-3.5.9/qtruby/rubylib/qtruby/Qt.cpp
===================================================================
--- kdebindings-3.5.9.orig/qtruby/rubylib/qtruby/Qt.cpp	2008-06-07 00:48:29.000000000 +0000
+++ kdebindings-3.5.9/qtruby/rubylib/qtruby/Qt.cpp	2008-06-07 00:50:23.000000000 +0000
@@ -950,7 +950,7 @@
 const char *
 get_VALUEtype(VALUE ruby_value)
 {
-	char * classname = rb_obj_classname(ruby_value);
+	const char * classname = rb_obj_classname(ruby_value);
     const char *r = "";
     if(ruby_value == Qnil)
 	r = "u";
@@ -1395,7 +1395,7 @@
 }
 
 static QCString *
-find_cached_selector(int argc, VALUE * argv, VALUE klass, char * methodName)
+find_cached_selector(int argc, VALUE * argv, VALUE klass, const char * methodName)
 {
     // Look in the cache
 static QCString * mcid = 0;
@@ -1432,7 +1432,7 @@
 static VALUE
 method_missing(int argc, VALUE * argv, VALUE self)
 {
-	char * methodName = rb_id2name(SYM2ID(argv[0]));
+	const char * methodName = rb_id2name(SYM2ID(argv[0]));
     VALUE klass = rb_funcall(self, rb_intern("class"), 0);
 
 	// Look for 'thing?' methods, and try to match isThing() or hasThing() in the Smoke runtime
@@ -1482,7 +1482,7 @@
 
 			VALUE retval = rb_funcall2(qt_internal_module, rb_intern("do_method_missing"), argc+3, temp_stack);
 			if (_current_method == -1) {
-				char * op = rb_id2name(SYM2ID(argv[0]));
+				const char * op = rb_id2name(SYM2ID(argv[0]));
 				if (	qstrcmp(op, "-") == 0
 						|| qstrcmp(op, "+") == 0
 						|| qstrcmp(op, "/") == 0
@@ -1557,7 +1557,7 @@
 class_method_missing(int argc, VALUE * argv, VALUE klass)
 {
 	VALUE result = Qnil;
-	char * methodName = rb_id2name(SYM2ID(argv[0]));
+	const char * methodName = rb_id2name(SYM2ID(argv[0]));
 	VALUE * temp_stack = (VALUE *) calloc(argc+3, sizeof(VALUE));
     temp_stack[0] = rb_str_new2("Qt");
     temp_stack[1] = rb_str_new2(methodName);
@@ -1772,7 +1772,7 @@
 VALUE
 getmetainfo(VALUE self, int &offset, int &index)
 {
-    char * signalname = rb_id2name(rb_frame_last_func());
+    const char * signalname = rb_id2name(rb_frame_last_func());
     VALUE metaObject_value = rb_funcall(qt_internal_module, rb_intern("getMetaObject"), 1, self);
 
     smokeruby_object *ometa = value_obj_info(metaObject_value);
